R/ops.R

```diff
@@ -123,10 +123,19 @@ st_apply.stars = function(X, MARGIN, FUN, ..., CLUSTER = NULL, PROGRESS = FALSE,
 				} else
 					parallel::parApply(CLUSTER, X = y, MARGIN = MARGIN, FUN = FUN, ...)
 			}
-		if (is.array(ret))
-			ret
-		else
-			array(ret, dX)
+		if (!is.array(ret))
+		{
+			ret <- array(ret,dX)
+		}
+		if(inherits(y,'units')){
+			# calculate function directly on one subset to see what the output units are
+			funUnits<-FUN(do.call('[',c( list(y),ifelse(1:length(dim(y)) %in% MARGIN,1,lapply(dim(y), seq)))))
+			if(inherits(funUnits,'units'))
+				if(all.equal(units(funUnits), units(y))){
```

Comment:
Why not just pass the units as they come: the function may change the units, e.g. if `FUN` is `log` or an integer power?

Reply:
I later realized that this would be the alternative and may be preferable option, I will change the request accordingly